webui/react/src/utils/step.ts

```diff
@@ -1,5 +1,41 @@
-import { CheckpointState, Step } from '../types';
+import { CheckpointState, CheckpointWorkload, MetricsWorkload, Step, Step2,
+  WorkloadWrapper } from '../types';
 
-export const hasCheckpoint = (step: Step): boolean => {
+export const hasCheckpointStep = (step: Step | Step2): boolean => {
   return !!step.checkpoint && step.checkpoint.state !== CheckpointState.Deleted;
 };
+
+export const hasCheckpoint = (workload: WorkloadWrapper): boolean => {
+  return !!workload.checkpoint && workload.checkpoint.state !== CheckpointState.Deleted;
+};
+
+export const getWorkload = (wrapper: WorkloadWrapper): MetricsWorkload | CheckpointWorkload => {
+  return Object.values(wrapper).find(val => !!val);
+};
+
+export const isMetricsWorkload = (workload: MetricsWorkload | CheckpointWorkload)
+: workload is MetricsWorkload => {
+  return 'metrics' in workload
+  && 'numInputs' in workload
+  && !('uuid' in workload)
+  && !('resources' in workload);
+};
+
+export const workloadsToSteps = (workloads: WorkloadWrapper[]): Step2[] => {
+  const stepsDict: Record<number, Partial<Step2>> = {};
+  workloads.forEach(wlWrapper => {
```

Comment:
**nit:** `workload` instead of `wlWrapper`?

Reply:
:thinking:  maybe I should have used wrapper in the original function parameter too. I add wrapper to be more explicit about this not being a `Workload` type

Reply:
how about renaming workloads to wlWrappers or wrappers? or workloadWrappers